src/match_identifiers.cpp

```diff
@@ -45,7 +45,7 @@ void match_identifiers(TypedAST::Identifier* ast, Frontend::CompileTimeEnvironme
 	}
 
 	// TODO: should this be a Var instead of just an id copy?
-	ast->m_value_type = ast->m_declaration->m_value_type;
+	ast->m_value_type = env.m_typechecker.m_core.inst_fresh(declaration->m_decl_type);
```

Comment:
Acceder al core desde afuera no es ideal, quizás habría que hacer una interfaz para esto

Reply:
This is basically the [Let] deduction rule, just lazily thrown together